main.tf

```diff
@@ -24,22 +25,23 @@ resource "azurerm_kubernetes_cluster" "main" {
   }
 
   default_node_pool {
-    name            = "nodepool"
-    node_count      = var.agents_count
-    vm_size         = var.agents_size
-    os_disk_size_gb = var.os_disk_size_gb
-    vnet_subnet_id  = var.vnet_subnet_id
+    orchestrator_version = var.kubernetes_version
```

Comment:
I'd like to suggest add one more variable for `orchestrator_version`, because they're not necessary to be the same.